LnkParse3/extra/metadata.py

```diff
@@ -46,45 +135,65 @@ def as_dict(self):
         tmp["storage_size"] = self.storage_size()
         tmp["version"] = self.version()
         tmp["format_id"] = self.format_id()
+        tmp["property_values"] = self.serialized_property_value()
         return tmp
 
-    # TODO:
     def serialized_property_value(self):
-        raise NotImplementedError
-        """
-        if _mpb['format_id'].upper() == 'D5CDD5052E9C101B939708002B2CF9AE':
+        start = 32
+        result = []
+        if self.format_id().upper() == 'D5CDD5052E9C101B939708002B2CF9AE':
             # Serialized Property Value (String Name)
-            index += 32
-            result = []
-
             while True:
-                value = {}
-                value['value_size'] = unpack('<I', self.indata[index: index + 4])[0]
+                value = dict()
+                value['value_size'] = unpack('<I', self._raw[start: start + 4])[0]
                 if hex(value['value_size']) == hex(0x0):
                     break
-                value['name_size'] = unpack('<I', self.indata[index + 4: index + 8])[0]
-                value['name'] = self.read_unicode_string(index + 8)
-                value['value'] = ''  # TODO MS-OLEPS
+                value['name_size'] = unpack('<I', self._raw[start + 4: start + 8])[0]
+                name_size_end = (start + 9 + int(value['name_size'])) * 2
+                binary = self._raw[start + 9: name_size_end]
+                value['name'] = self.text_processor.read_unicode_string(binary)
+
+                value_type = unpack('<H', self._raw[name_size_end: name_size_end + 2])[0]
+                value_padding = unpack('<H', self._raw[name_size_end + 2: name_size_end + 4])[0]
+                if value_padding == 0:
+                    match value_type:
+                        case PropertyType.VT_I2:
+                            value['value'] = unpack('<I', self._raw[name_size_end + 4: name_size_end + 6])[0]
```

Comment:
I think this code should be either removed or fixed (including the similar commented code in the `else` branch). Correct me if I am wrong, but there should be unpacking of signed integer instead of unsigned integer (`I` -> `i`).

Reply:
```suggestion
                        case PropertyType.VT_I2:
                            value['value'] = unpack('<i', self._raw[name_size_end + 4: name_size_end + 6])[0]
```